src/pg/sql/41_observatory_augmentation.sql

```diff
@@ -0,0 +1,995 @@
+
+--For Longer term Dev
+
+
+--Break out table definitions to types
+--Automate type creation from a script, something like
+----CREATE OR REPLACE FUNCTION OBS_Get<%=tag_name%>(geom GEOMETRY)
+----RETURNS TABLE(
+----<%=get_dimensions_for_tag(tag_name)%>
+----AS $$
+----DECLARE
+----target_cols text[];
+----names text[];
+----vals numeric[];-
+----q text;
+----BEGIN
+----target_cols := Array[<%=get_dimensions_for_tag(tag_name)%>],
+
+
+--Functions for augmenting specific tables
+--------------------------------------------------------------------------------
+
+-- Creates a table of demographic snapshot
+-- TODO: Remove since it does address geocoding?
+CREATE OR REPLACE FUNCTION OBS_GetDemographicSnapshot(address text)
+RETURNS TABLE(
+  total_pop NUMERIC,
+  male_pop NUMERIC,
+  female_pop NUMERIC,
+  median_age NUMERIC,
+  white_pop NUMERIC,
+  black_pop NUMERIC,
+  asian_pop NUMERIC,
+  hispanic_pop NUMERIC,
+  amerindian_pop NUMERIC,
+  other_race_pop NUMERIC,
```

Comment:
No. The correct way to handle this would be to pre-process and generate the function signature from metadata, since there's no way to alter the columns returned at runtime.

Essentially, these columns all correspond to the census demography tag, and the `obs_column_tag`, `obs_column` and `obs_column_table` tables can generate this signature.